pkg/reconciler/pubsub/reconciler.go

```diff
@@ -18,221 +18,145 @@ package pubsub
+		return t, nil, fmt.Errorf("Topic %q not ready", t.Name)
 	}
 
-	c.Logger.Debug("Job still active.", zap.Any("job", job))
-	return ops.OpsJobOngoing, nil
-}
-
-func (c *PubSubBase) ensureSubscriptionJob(ctx context.Context, args operations.SubArgs) (ops.OpsJobStatus, error) {
-	job, err := c.getJob(ctx, args.Owner.GetObjectMeta(), labels.SelectorFromSet(operations.SubscriptionJobLabels(args.Owner, args.Action)))
-	// If the resource doesn't exist, we'll create it
-	if apierrs.IsNotFound(err) {
-		c.Logger.Debugw("Job not found, creating.")
-
-		args.Image = c.SubscriptionOpsImage
-
-		job = operations.NewSubscriptionOps(args)
-
-		job, err := c.KubeClientSet.BatchV1().Jobs(args.Owner.GetObjectMeta().GetNamespace()).Create(job)
-		if err != nil || job == nil {
-			c.Logger.Errorw("Failed to create Job.", zap.Error(err))
-			return ops.OpsJobCreateFailed, err
-		}
-
-		c.Logger.Debugw("Created Job.")
-		return ops.OpsJobCreated, nil
-	} else if err != nil {
-		c.Logger.Errorw("Failed to get Job.", zap.Error(err))
-		return ops.OpsJobGetFailed, err
+	if t.Status.ProjectID == "" {
+		status.MarkTopicNotReady(cs, "TopicNotReady", "Topic %q did not expose projectid", t.Name)
+		return t, nil, fmt.Errorf("Topic %q did not expose projectid", t.Name)
 	}
 
-	if ops.IsJobFailed(job) {
-		c.Logger.Debugw("Job has failed.")
-		var sar operations.SubActionResult
-		if err := c.UnmarshalJobTerminationMessage(ctx, job.Namespace, job.Name, &sar); err != nil {
-			c.Logger.Debugw("Failed to unmarshal termination message.", zap.Error(err))
-			return ops.OpsJobCompleteFailed, errors.New(ops.JobFailedMessage(job))
-		}
-		return ops.OpsJobCompleteFailed, errors.New(sar.Reason)
+	if t.Status.TopicID == "" {
+		status.MarkTopicNotReady(cs, "TopicNotReady", "Topic %q did not expose topicid", t.Name)
+		return t, nil, fmt.Errorf("Topic %q did not expose topicid", t.Name)
 	}
 
-	if ops.IsJobComplete(job) {
-		c.Logger.Debugw("Job is complete.")
-		return ops.OpsJobCompleteSuccessful, nil
+	if t.Status.TopicID != topic {
+		status.MarkTopicNotReady(cs, "TopicNotReady", "Topic %q mismatch: expected %q got %q", t.Name, topic, t.Status.TopicID)
+		return t, nil, fmt.Errorf("Topic %q mismatch: expected %q got %q", t.Name, topic, t.Status.TopicID)
 	}
 
-	c.Logger.Debug("Job still active.", zap.Any("job", job))
-	return ops.OpsJobOngoing, nil
-}
+	status.TopicID = t.Status.TopicID
+	status.ProjectID = t.Status.ProjectID
+	status.MarkTopicReady(cs)
 
-// UnmarshalJobTerminationMessage unmarshals the job pod termination message.
-func (c *PubSubBase) UnmarshalJobTerminationMessage(ctx context.Context, namespace, name string, v interface{}) error {
-	jobPod, err := ops.GetJobPodByJobName(ctx, c.KubeClientSet, namespace, name)
+	// Ok, so the Topic is ready, let's reconcile PullSubscription.
+	pullSubscriptions := psb.pubsubClient.PubsubV1alpha1().PullSubscriptions(namespace)
+	ps, err := pullSubscriptions.Get(name, v1.GetOptions{})
```

Comment:
nit: seems no need to have two lines.

Reply:
needed here. But not in the delete. Thanks for the catch!